.github/workflows/release.yml

```diff
@@ -185,13 +181,41 @@ jobs:
           path: |
             ./tmp/${{ env.artifact_name }}.tgz
 
+      - name: Generate Release Notes
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          new_commit_hash=$(git rev-parse HEAD)
+          echo "New commit hash: $new_commit_hash"
+
+          previous_tag=$(git tag --sort=-version:refname | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+" | grep -Fv "${{ env.iota_ref }}" | head -1)
+          echo "Previous tag: $previous_tag"
+          previous_commit_hash=$(git rev-list -n 1 "${previous_tag}")
+          echo "Previous commit hash: $previous_commit_hash"
+
+          pushd "scripts/release_notes"
+          ./run.sh generate $previous_commit_hash $new_commit_hash | tee -a "../../${{ env.RELEASE_NOTES_FILE }}"
+          popd
+          echo "---" >> ${{ env.RELEASE_NOTES_FILE }}
+          echo "#### Full Log: https://github.com/iotaledger/iota/commits/${{ env.iota_ref }}" >> ${{ env.RELEASE_NOTES_FILE }}
+
+          cat ${{ env.RELEASE_NOTES_FILE }}
+
+          if [[ ${{ env.iota_ref }} == devnet* ]]; then
```

Comment:
I don't get the `devnet*` check here

Reply:
This is to set the pre-release flag on Devnet releases only 
<img width="93" alt="image" src="https://github.com/user-attachments/assets/e21abb3d-0ebe-4e42-9560-31e175a93803" />

Reply:
But I'm not too sure why it's needed now. Were we setting it manually?

Reply:
it is set manually

Reply:
Then this is automating it

Reply:
But `devnet` is not part of the `iota_ref`, so this makes no sense

Reply:
I guess she meant `contains(github.ref, 'devnet')` or something like that. What do you think would be best here?